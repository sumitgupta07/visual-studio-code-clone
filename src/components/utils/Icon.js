import React from 'react';
import './Icon.css';

const Icon = ({ label, onSelectIcon, iconClass }) => {
  return (
    <button onClick={() => onSelectIcon(label)}>
      <div className="tooltip">
        <div className="icon">{<i className={iconClass}></i>}</div>
        <div className="tooltiptext">
          <span>{label}</span>
        </div>
      </div>
    </button>
  );
};

export default Icon;
