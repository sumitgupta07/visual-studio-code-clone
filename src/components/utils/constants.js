export const labels = {
  FileExplorer: 'File Explorer',
  Search: 'Search',
  SourceControl: 'Source Control',
  Debug: 'Debug',
  Extensions: 'Extensions'
};

export const classNames = {
  explorerClass: 'far fa-copy fa-3x',
  searchClass: 'fas fa-search fa-rotate fa-3x',
  sourcecontrolClass: 'fas fa-code-branch fa-3x',
  debugClass: 'fas fa-bug fa-3x',
  extensionsClass: 'fas fa-th-large fa-3x'
};

export const folders = [
  {
    id: 1,
    name: 'src',
    files: [
      {
        id: 'a',
        fileName: 'index.js',
        content: 'Placeholder for index.js. It has App.js as child components'
      },
      {
        id: 'b',
        fileName: 'App.js',
        content:
          'Placeholder for App.js. It has Header and Content as child components'
      },
      {
        id: 'c',
        fileName: 'Content.js',
        content:
          'Placeholder for Content.js file. It has Navbar, SelectPane and EditorPane as child components'
      }
    ]
  }
];
