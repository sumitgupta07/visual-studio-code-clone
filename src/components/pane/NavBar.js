import React from 'react';
import PropTypes from 'prop-types';
import Icon from '../utils/Icon';
import { labels, classNames } from '../utils/constants';

const NavBar = ({ onSelectNav }) => {
  const handleSelectIcon = item => {
    onSelectNav(item);
  };

  return (
    <div className="sidebar">
      <Icon
        id="FileExplorer"
        label={labels.FileExplorer}
        iconClass={classNames.explorerClass}
        onSelectIcon={handleSelectIcon}
      />
      <Icon
        id="Search"
        label={labels.Search}
        iconClass={classNames.searchClass}
        onSelectIcon={handleSelectIcon}
      />
      <Icon
        id="SourceControl"
        label={labels.SourceControl}
        iconClass={classNames.sourcecontrolClass}
        onSelectIcon={handleSelectIcon}
      />
      <Icon
        id="Debug"
        label={labels.Debug}
        iconClass={classNames.debugClass}
        onSelectIcon={handleSelectIcon}
      />
      <Icon
        id="Extensions"
        label={labels.Extensions}
        iconClass={classNames.extensionsClass}
        onSelectIcon={handleSelectIcon}
      />
    </div>
  );
};
NavBar.propTypes = {
  onSelectNav: PropTypes.func.isRequired
};
export default NavBar;
