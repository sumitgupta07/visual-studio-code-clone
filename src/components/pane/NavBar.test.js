import React from 'react';
import NavBar from './NavBar';
import { mount } from 'enzyme';
import { labels } from '../utils/constants';

let handleSelect;
let wrapper;
let mockProps;
beforeEach(() => {
  handleSelect = jest.fn();
  mockProps = {
    onSelectNav: handleSelect
  };
  wrapper = mount(<NavBar {...mockProps} />);
});

it('should render NavBar component', () => {
  expect(wrapper).toMatchSnapshot();
});
it('should pass Search as label for Icon with id: Search', () => {
  expect(wrapper.find('#Search').prop('label')).toEqual('Search');
});

it('should call onSelectNav prop with parameter "FileExplorer"', () => {
  wrapper
    .find('button')
    .first()
    .simulate('click');
  expect(handleSelect).toHaveBeenCalledWith(labels.FileExplorer);
});

it('should call onSelectNav prop with parameter "Debug"', () => {
  wrapper
    .find('#Debug')
    .find('button')
    .simulate('click');
  expect(handleSelect).toHaveBeenCalledWith(labels.Debug);
});
