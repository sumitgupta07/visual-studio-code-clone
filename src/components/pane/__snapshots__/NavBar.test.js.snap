// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render NavBar component 1`] = `
<NavBar
  onSelectNav={[MockFunction]}
>
  <div
    className="sidebar"
  >
    <Icon
      iconClass="far fa-copy fa-3x"
      id="FileExplorer"
      label="File Explorer"
      onSelectIcon={[Function]}
    >
      <button
        onClick={[Function]}
      >
        <div
          className="tooltip"
        >
          <div
            className="icon"
          >
            <i
              className="far fa-copy fa-3x"
            />
          </div>
          <div
            className="tooltiptext"
          >
            <span>
              File Explorer
            </span>
          </div>
        </div>
      </button>
    </Icon>
    <Icon
      iconClass="fas fa-search fa-rotate fa-3x"
      id="Search"
      label="Search"
      onSelectIcon={[Function]}
    >
      <button
        onClick={[Function]}
      >
        <div
          className="tooltip"
        >
          <div
            className="icon"
          >
            <i
              className="fas fa-search fa-rotate fa-3x"
            />
          </div>
          <div
            className="tooltiptext"
          >
            <span>
              Search
            </span>
          </div>
        </div>
      </button>
    </Icon>
    <Icon
      iconClass="fas fa-code-branch fa-3x"
      id="SourceControl"
      label="Source Control"
      onSelectIcon={[Function]}
    >
      <button
        onClick={[Function]}
      >
        <div
          className="tooltip"
        >
          <div
            className="icon"
          >
            <i
              className="fas fa-code-branch fa-3x"
            />
          </div>
          <div
            className="tooltiptext"
          >
            <span>
              Source Control
            </span>
          </div>
        </div>
      </button>
    </Icon>
    <Icon
      iconClass="fas fa-bug fa-3x"
      id="Debug"
      label="Debug"
      onSelectIcon={[Function]}
    >
      <button
        onClick={[Function]}
      >
        <div
          className="tooltip"
        >
          <div
            className="icon"
          >
            <i
              className="fas fa-bug fa-3x"
            />
          </div>
          <div
            className="tooltiptext"
          >
            <span>
              Debug
            </span>
          </div>
        </div>
      </button>
    </Icon>
    <Icon
      iconClass="fas fa-th-large fa-3x"
      id="Extensions"
      label="Extensions"
      onSelectIcon={[Function]}
    >
      <button
        onClick={[Function]}
      >
        <div
          className="tooltip"
        >
          <div
            className="icon"
          >
            <i
              className="fas fa-th-large fa-3x"
            />
          </div>
          <div
            className="tooltiptext"
          >
            <span>
              Extensions
            </span>
          </div>
        </div>
      </button>
    </Icon>
  </div>
</NavBar>
`;
